code/modules/mob/living/carbon/xenomorph/castes/Queen.dm

```diff
@@ -792,13 +792,24 @@
 		return TRUE
 
 /mob/living/carbon/xenomorph/queen/death(cause, gibbed)
-	if(hive.living_xeno_queen == src)
+	if(src == hive?.living_xeno_queen)
 		hive.xeno_queen_timer = world.time + XENO_QUEEN_DEATH_DELAY
-		hive.banished_ckeys   = list() // Reset the banished ckey list
+
+		// Reset the banished ckey list
+		if(length(hive.banished_ckeys))
+			for(var/mob/living/carbon/xenomorph/target_xeno in hive.totalXenos)
```

Comment:
You know more than me, but why can't we as anything?

Reply:
It's not that we can't, but this proc is called 0-a few times ever in a round so we might as well enforce the variables contained in the list are properly typed. In other words this is very cold code that is not important to be as performant as possible. As anything means we skip the type checking; but not doing it on a typed iterator means we won't risk a runtime should there be an entry that doesn't cast to that type.